src/porepy/applications/md_grids/model_geometries.py

```diff
@@ -155,3 +155,96 @@ def meshing_arguments(self) -> dict:
+
+        We here make a non-matching mixed-dimensional grid in the sense that neither of
+        the fracture grids, interface grids or the rock grid are matching. This is done
+        by refining the fracture grids and interfaces.
+
+        The fracture grids are replaced by a refined version of the fractures which are
+        already present in the grid. The interface grids are replaced by first creating
+        a new mixed-dimensional grid with a higher resolution. The new mixed-dimensional
+        grid will "donate" its interface grids to the original mixed-dimensional grid.
+
+        There are a few things to be aware of when creating a non-matching
+        mixed-dimensional grid:
+            * If _both_ the interface and fracture grids are to be refined/coarsened,
+              you must be aware of:
+                * If you first refine (and replace) the fracture grids, you must also
+                  update the fracture numbering such that the new fracture grids have
+                  the same fracture number as the old ones. This is because the fracture
+                  numbering is used to identify the correct interface grids later.
+                  TODO: Make sure that the refine-grid-functions also copy frac_num
+                  attribute such that the abovementioned is no longer a problem.
+            * Ensure that the "donor" mixed-dimensional grid is physically the same as
+              the "recipient" mixed-dimensional grid. Fractures must be located at the
+              same place and the physical dimension of the grids must be the same.
+
+        """
+        super().set_geometry()
+
+        # Refine and replace fracture grids:
+        old_fracture_grids = self.mdg.subdomains(dim=1)
+
+        # Ratios which we want to refine the fracture grids with.
+        ratios = []
+        for i in range(len(old_fracture_grids)):
+            ratios.append(i + 2)
+
+        new_fracture_grids = [
+            pp.refinement.refine_grid_1d(g=old_grid, ratio=ratio)
+            for old_grid, ratio in zip(old_fracture_grids, ratios)
+        ]
+
+        grid_map = dict(zip(old_fracture_grids, new_fracture_grids))
+
+        # Refine and replace interface grids:
+        # We first create a new and more refined mixed-dimensional grid.
+        def mdg_func(self, nx=2, ny=2) -> pp.MixedDimensionalGrid:
+            """Generate a refined version of an already existing mixed-dimensional grid.
+
+            Parameters:
+                nx: Number of cells in x-direction.
+                ny: Number of cells in y-direction.
+
+            """
+            fracs = [self.fractures[i].pts for i in range(len(self.fractures))]
+            domain = self.domain.bounding_box
+            md_grid = pp.meshing.cart_grid(
+                fracs, np.array([nx, ny]), physdims=[domain["xmax"], domain["ymax"]]
+            )
+            return md_grid
+
+        mdg_new = mdg_func(self, nx=6, ny=6)
```

Comment:
Two questions: 
1) Why not formulate this as refinement (so a ratio), rather than cell number? This will align better with the approach taken for the fracture grids.
2) Can the refinement level become a user controlled parameter?

Reply:
I addressed these two things in a commit just now. Would you have a quick look whenever you have time to see if you agree with closing this conversation? My sketch of 1. feels a bit brute force... 

Also addressed the fracture refinement ratios being controlled by a user parameter, so that conversation might be closed soon too.